apis/Google.Cloud.Logging.NLog/Google.Cloud.Logging.NLog/GoogleCloudLoggingTarget.cs

```diff
@@ -0,0 +1,389 @@
+// Copyright 2018 Google Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Security;
+using Google.Api;
+using Google.Api.Gax;
+using Google.Api.Gax.Grpc;
+using Google.Apis.Auth.OAuth2;
+using Google.Cloud.Logging.Type;
+using Google.Cloud.Logging.V2;
+using Google.Protobuf;
+using Google.Protobuf.WellKnownTypes;
+using NLog;
+using NLog.Config;
+using NLog.Targets;
+using NLog.Common;
+using NLog.Layouts;
+using System.Threading.Tasks;
+using System.Threading;
+
+namespace Google.Cloud.Logging.NLog
+{
+    /// <summary>
+    /// 
+    /// </summary>
+    [Target("GoogleCloud")]

```

Comment:
We'd really only what this ever used with the/an async wrapper; you've probably seen the Log4Net code essentially implements async itself.
Is there any way to enforce using the async wrapper?
What's the NLog idiom for targets that should only be used with the async wrapper?

Reply:
Would think that `_client.WriteLogEntriesAsync(...)' was a non-blocking operation, so should not matter whether  async-wrapper is used. But there is no way to force one to use async-wrapper, besides recommendation in Wiki.

Reply:
It's the buffering behaviour that I'm more interested in.
In a application with a lot of logging, we do not want every log statement in the application to lead to an API call to stackdriver.
Having read the async-wrapper docs, it looks like the async-wrapper will provide this buffering functionality. Please can you confirm if this is correct? Thanks.

Reply:
Yes the async-wrapper will enable batching of logevents, so they arrive in chunks instead of individual logevents.